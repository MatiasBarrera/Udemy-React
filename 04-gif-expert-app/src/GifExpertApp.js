import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import GifGrid from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = []}) => {
  // const [categories, setCategories] = useState(['Hunter x Hunter'])
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory setCategories={setCategories}/>
      <hr/>
      {
        categories.map(category =>
          <GifGrid 
            key={category}
            category={category} 
          />)
      }
    </>
  )
}
