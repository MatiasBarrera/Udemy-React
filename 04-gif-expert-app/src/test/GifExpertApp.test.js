import { shallow } from "enzyme";
import {GifExpertApp} from '../GifExpertApp'

describe('Prueba de componente GifExpertApp.js', () => {
  test('Debe de mostrar componente <GifExpertApp /> correctamente', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  });

  test('Debe de mostrar una lista de categorias', () => {
    const categories = ['hunter x', 'dragon ball']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  });
});