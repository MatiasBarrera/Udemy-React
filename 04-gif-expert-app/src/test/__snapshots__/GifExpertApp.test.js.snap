// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifExpertApp.js Debe de mostrar componente <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Prueba de componente GifExpertApp.js Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="hunter x"
    key="hunter x"
  />
  <GifGrid
    category="dragon ball"
    key="dragon ball"
  />
</Fragment>
`;
